### PR TITLE
makefile - add bin_SCRIPTS

### PR DESCRIPTION
**Pull request by hppritcha**
At some point there may be a use for adding scripts to bin
as part of install, as well as to have nodist directories.
By setting these to empty in the top level makefile, these
variables can be extended within submakefiles for provider
or component specific scripts, etc.

Updates ofi-cray/libfabric-cray#531

@shefty 
@goodell 
@sungeunchoi 

Signed-off-by: Howard Pritchard <howardp@lanl.gov>

### CODE REVIEW

File: Makefile.am [hunk 1]
```diff
@@ -40,6 +40,10 @@ linkback = $(top_builddir)/src/libfabric.la
 bin_PROGRAMS = \
 	util/fi_info
 
+bin_SCRIPTS =
+
+nodist_DIR =
```
Comment (goodell) [inline Makefile.am]: I'm not sure that `nodist_DIR` makes sense (i.e., I don't think that automake supports a `DIR` variable to be `nodist`ed).  Do you have a companion commit over in the Cray/GNI development repository that I can look at to see how you're using this?  I have different concerns about dropping `dist_` from `dist_bin_SCRIPTS` that will probably also be cleared up by seeing the companion commit.

**Comment by hppritcha** [comment]
@goodell sorry for delay in getting back on this, see fb14e73 for the motivator.

**Comment by goodell** [comment]
OK, after looking at both commits together, here's my opinion:

1. You're abusing `nodist_` a bit, so we end up with a distribution tarball that is no longer build-able when the user passes `--with-criterion=DIR`.  I know this is sort of intentional, but it's icky.  Adding a small amount of logic to `prov/gni/configure.m4` to disable the `HAVE_CRITERION` automake conditional when `${top_srcdir}/prov/gni/test/common.c` is absent would lead to a better user experience in this corner case.

2. I still think `nodist_DIR` is nonsense.  Do you actually get a behavior change when you use it or don't use it?  A quick test on a skeleton autotools project on my machine indicates it doesn't change anything.  In the absence of noticeable behavior change, strike it from both commits.

3. (you may already be doing this) Double check that `make distcheck` works with your modifications.  It should, but this is probably the best test we have besides eyeballing it.  Un-tar the resulting archive and check that you don't get a `prov/gni/test` directory.

**Comment by hppritcha** [comment]
@goodell thanks for the feedback.  As for 1) you're right.  But that's a provider specific thing that we'll fix in the provider configury. 2) I checked and you're right there.  The nodist_DIR is not needed to keep the gni provider criterion tests and the test dir to show up in the dist tarball. I'll redo the pr without a nodist_DIR variable.  3) that's how I was checking that I was in fact addressing ofi-cray/libfabric-cray#531

From the perspective of other providers, the main thing the gni provider folks don't want to do is mess up the configury/automake part of the project by squashing/resetting an automake or autoconf variable.

**Comment by hppritcha** [comment]
pushed new commits - no more nodist_DIR.
